Updating TN0010: Designing response types (#601)
apollographql/docs: src/content/technotes/TN0010-reponse-type-pattern.mdx

* If the list is empty, is that because there are zero users, or did an error occur?
* Even if the list is populated, did the API return _all_ users or just a subset?
* Are there multiple pages of results?

> To answer questions like these, it may be helpful for top-level fields of `Query` to return "wrapper" objects that can include both the operation result _and_ metadata about the operation's execution. Some examples of where this is specifically helpful is in cases where you need to [paginate your results](/technotes/TN0027-demand-oriented-schema-design/#a-brief-sidebar-on-pagination-conventions) or implement [Relay-style connections](/technotes/TN0029-relay-style-connections/).

Reviewer: ```suggestion
To answer questions like these, it may be helpful for top-level fields of `Query` to return "wrapper" objects that can include both the operation result _and_ metadata about the operation's execution. For example, in cases where you need to [paginate your results](/technotes/TN0027-demand-oriented-schema-design/#a-brief-sidebar-on-pagination-conventions) or implement [Relay-style connections](/technotes/TN0029-relay-style-connections/) this pattern can be helpful.
```

Author: Thanks, updated!

### Example: `UsersResponse`

The following example defines a `UsersResponse` type for our `Query.users` field:

```graphql
type User {
  id: ID!
  firstName: String!
  lastName: String!
}

type UsersResponse {
  offset: Int!

>   perPage: Int!

Reviewer: minor nit- most folks use a `limit` value here when doing offset-based pagination, but understand why that's not necessarily the clearest doc example. :)

Author: Thanks, updated!
